backend/entityservice/cache/active_runs.py

```diff
@@ -0,0 +1,49 @@
+"""
+Cache information about the run state in redis.
+
+Note across the cache we use a common redis hash for each run,
+this module uses the 'state' key.
+"""
+from enum import Enum
+import structlog
+
+from entityservice.cache.connection import connect_to_redis
+from entityservice.cache.helpers import _get_run_hash_key
+
+logger = structlog.get_logger()
+
+
+class RunState(Enum):
+    MISSING = None
+    ACTIVE = b'active'
+    DELETED = b'deleted'
```

Comment:
Need documentation about these states: what is an active run? Does that mean it is running? What about finished runs?

Reply:
Sure I'll add a docstring but I'll point out this isn't exposed outside of this file - it is simply to support the public api of: `is_run_active`, `set_run_state_active` and `set_run_state_deleted`.

We could add (and expose) states like "finished" - if and when required.

Reply:
I was also wondering yesterday, should we include an `ERROR` state?

Reply:
Perhaps at some point, but I don't want to prematurely add things to the cache that are not used.